Scala/fasta/fasta_idiomatic.scala

```diff
@@ -0,0 +1,110 @@
+/* The Computer Language Benchmarks Game
+   http://benchmarksgame.alioth.debian.org/
+  based on original contributed by Isaac Gouy
+  updated for 2.9 and optimized by Rex Kerr
+ */
+
+import java.io._
+
+object fasta {
```

Comment:
It is making IO however it also can be a good measurement for Streaming APIs (Lazy List). What we need to do it either: 
* print repeated sequence taking N elements times with given line length 
* print sequence of N elements by using selected randomly elements with given line length
where N is in benchmark equal to 25.000.000 * size of input sequence.
The most idiomatic way is here usage of LazyLists:  
```
   LazyList
      .continually(sequence)
      .flatten
      .take(length)
      .grouped(LineLength)
      .foreach { line =>
        write(line.toArray)
        write(EOL)
      }
```
However, this solution is at least 2x slower than using of custom while loop.

Reply:
I also tried a version with parallel collections, especially for the 2nd case to select elements in parallel, however, it does not work well, probably, because it needs to know the size of the collection, which for 750M (N=25M * size of sequence) elements leads to OutOfMemoery errors.

Reply:
OK to keep it as it is.